File: presence/concept-pl-execution-13-fixtures/concept_pulse.swift
// concept: selective-retention
let signals = [2, 7, 1, 8, 2, 8]
let threshold = 5
let scoreFloor = 100
let retainedFloor = 3
var score = 0
var retained = 0
for (index, value) in signals.enumerated() {
  score += value * (index + 1)
  if value >= threshold { retained += 1 }
}
let verdict = score >= scoreFloor && retained >= retainedFloor ? "coherent" : "fragmented"
print("\(verdict):\(score):\(retained)")

File: presence/concept-pl-execution-13-fixtures/ConceptPulse.kt
// concept: selective-retention
fun main() {
  val signals = listOf(2, 7, 1, 8, 2, 8)
  val threshold = 5
  val scoreFloor = 100
  val retainedFloor = 3
  var score = 0
  var retained = 0
  signals.forEachIndexed { index, value ->
    score += value * (index + 1)
    if (value >= threshold) retained++
  }
  val verdict = if (score >= scoreFloor && retained >= retainedFloor) "coherent" else "fragmented"
  println("$verdict:$score:$retained")
}
